packages/grid/_modules_/grid/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { optionsSelector } from '../../utils/optionsSelector';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useGridSelector } from '../core/useGridSelector';
+import { gridContainerSizesSelector } from '../../../components/GridViewport';
+import { useNativeEventListener } from '../../root/useNativeEventListener';
+import { GRID_SCROLL } from '../../../constants/eventsConstants';
+
+export const useGridInfiniteLoader = (
+  windowRef: React.MutableRefObject<HTMLDivElement | null>,
+  apiRef: GridApiRef,
+): void => {
+  const options = useGridSelector(apiRef, optionsSelector);
+  const containerSizes = useGridSelector(apiRef, gridContainerSizesSelector);
```

Comment:
This would be a lot clearer, IHMO

```suggestion
  const containerSizes = useGridSelector(apiRef, (state: GridState) => state.containerSizes);
```

Reply:
not really as if we change the state structure we have to rewrite every function that use the containerSizes.

Reply:
The advantage is to make `state.containerSizes` something you can grep in the whole codebase for 1. refactoring 2. understanding the logic. Right now, a developer, needs to grep state.containerSizes **and**  gridContainerSizesSelector.

Reply:
Well if you grep or find usages, it will appear in less places. So if you need to refactor the state, you will refactor the selector and it will work in the entire codebase.

Reply:
>  reducing our public API

Why is `gridContainerSizesSelector` public? https://codesandbox.io/s/material-demo-forked-lv2zx?file=/demo.tsx This sounds completely wrong.

Reply:
Well, these values are coming from the state which is public. I can remove the selector all together but you could still get those values.

Reply:
I don't understand why it is an issue. Most ppl won't import it. And if they do, it's practical.

Reply:
The more modules we expose, the more likely they will get in the way of doing a refactorization. This could mean we can't ship a feature without waiting 1 year for the next BCs window.